peerconnection.go

```diff
@@ -400,6 +407,41 @@ func (pc *PeerConnection) CreateOffer(options *OfferOptions) (SessionDescription
 		return SessionDescription{}, &rtcerr.InvalidStateError{Err: ErrConnectionClosed}
 	}
 
+	isPlanB := pc.configuration.SDPSemantics == SDPSemanticsPlanB
+	if pc.currentRemoteDescription != nil {
+		isPlanB = descriptionIsPlanB(pc.RemoteDescription())
+	}
+
+	// include unmatched local transceivers
+	if !isPlanB {
+		// update the greater mid if the remote description provides a greater one
+		if pc.currentRemoteDescription != nil {
+			for _, media := range pc.currentRemoteDescription.parsed.MediaDescriptions {
+				mid := getMidValue(media)
+				if mid == "" {
+					continue
+				}
+				numericMid, err := strconv.Atoi(mid)
+				if err != nil {
+					continue
+				}
+				if numericMid > pc.greaterMid {
+					pc.greaterMid = numericMid
+				}
+			}
+		}
+		for _, t := range pc.GetTransceivers() {
+			if t.Mid() != "" {
+				continue
+			}
+			pc.greaterMid++
+			err := t.setMid(strconv.Itoa(pc.greaterMid))
+			if err != nil {
+				return SessionDescription{}, err
+			}
+		}
+	}
+
```

Comment:
I think this should happen in `SetLocalDescription` and not in `CreateOffer`?

> ```
> 5.9.  Applying a Local Description
> 
>    [...]
> 
>    o  If the m= section proto value indicates use of RTP:
> 
>       *  If there is no RtpTransceiver associated with this m= section,
>          find one and associate it with this m= section according to the
>          following steps.  Note that this situation will only occur when
>          applying an offer.
> 
>          +  Find the RtpTransceiver that corresponds to this m= section,
>             using the mapping between transceivers and m= section
>             indices established when creating the offer.
> 
>          +  Set the value of this RtpTransceiver's mid property to the
>             MID of the m= section.
> ```

Reply:
@sgotti and I discussed this on Slack and decided to make this change in a later PR to keep this one simpler.